### PR TITLE
Add tests for TimeZoneInfo to ensure offset is not outside of range

### PR DESCRIPTION
**Pull request by krwq**
Test for https://github.com/dotnet/coreclr/pull/18305
This requires the PR to get merged to pass on all platforms (currently merged but waiting for new coreclr build to be picked up by corefx)

### CODE REVIEW

**Comment by stephentoub** [comment]
> This requires the PR to get merged to pass on all platforms

Why is it passing CI without that?

File: src/System.Runtime/tests/TimeZoneInfoTests.cs [hunk 1]
```diff
@@ -274,6 +274,53 @@ public static void ToSerializedString_FromSerializedString_RoundTrips(TimeZoneIn
...
+            foreach (TimeZoneInfo tzi in TimeZoneInfo.GetSystemTimeZones())
+            {
+                foreach (TimeZoneInfo.AdjustmentRule ar in tzi.GetAdjustmentRules())
+                {
+                    Assert.True(Math.Abs((ar.DaylightDelta + tzi.BaseUtcOffset).TotalHours) <= 14.0);
+                }
+            }
+        }
+
+        static TimeZoneInfo TryGetSystemTimeZone(string id)
```
Comment (stephentoub) [inline src/System.Runtime/tests/TimeZoneInfoTests.cs]: Nit: private

**Comment by krwq** [comment]
seems like there were two problems here:
- runs did not show (or run) the test
- the tests were not failing because TimeZone was not set to Asia/Manila - I have addressed this issue

**Comment by krwq** [comment]
test this please

**Pull request merged**